app/Http/Controllers/ClassesController.php

```diff
@@ -77,9 +110,102 @@ public function create()
+        $DayHeldOn = null;
+        $CreateClassDates = false;
+        $CreateClassDays = false;
+        $DaysArray = array();
+
+        if ($SeasonType == '1') {
+          if ($request['monday']) {
+            $DaysArray[] = 'Monday';
+          }
+          if ($request['tuesday']) {
+            $DaysArray[] = 'Tuesday';
+          }
+          if ($request['wednesday']) {
+            $DaysArray[] = 'Wednesday';
+          }
+          if ($request['thursday']) {
+            $DaysArray[] = 'Thursday';
+          }
+          if ($request['friday']) {
+            $DaysArray[] = 'Friday';
+          }
+          if ($request['saturday']) {
+            $DaysArray[] = 'Saturday';
+          }
+          if ($request['sunday']) {
+            $DaysArray[] = 'Sunday';
+          }
+          $CreateClassDays = true;
+        } else {
+          $CreateClassDates = true;
+        }
+        $request->offsetUnset('monday');
+        $request->offsetUnset('tuesday');
+        $request->offsetUnset('wednesday');
+        $request->offsetUnset('thursday');
+        $request->offsetUnset('friday');
+        $request->offsetUnset('saturday');
+        $request->offsetUnset('sunday');
+        $request->offsetUnset('DayHeldOn');
+
+        $StartTime = $request['selectedHour'] . ":" . $request['selectedMinute'] . " " . $request['selectedAMPM'];
+        $StartTime = DateTime::createFromFormat('H:i A', $StartTime);
+        $StartTime = $StartTime->format('H:i:s');
+        $request->offsetUnset('selectedHour');
+        $request->offsetUnset('selectedMinute');
+        $request->offsetUnset('selectedAMPM');
+
+        $Length = $request['selectedHourLength'] * 60 + $request['selectedMinuteLength'];
+        $request->offsetUnset('selectedHourLength');
+        $request->offsetUnset('selectedMinuteLength');
+        
+        $selectedDates = explode(",", $request['selectedDates']);
+        $request->offsetUnset('selectedDates');
+
+        $class = new Classes;
+        $class->fill($request->all());
+        $class->season_id = $season_id;
+        if ($SeasonType == '1') {
+          $class->StartTime = $StartTime;
+          $class->Length = $Length;
+        } else {
+          $class->StartTime = null;
+          $class->Length = null;
+        }
+        $class->created_by = Auth::id();
+        $class->created_at = date('Y-m-d H:i:s');
+        $class->save();
+        if ($CreateClassDates) {
+          foreach ($selectedDates as $date) {
+            $classdate = new ClassDate;
+            $classdate->classes_id = $class->id;
+            $classdate->HeldOn = date('Y-m-d H:i:s', strtotime($date));
+            $classdate->save();
+          }
+        }
+
+        if ($CreateClassDays) {
+          foreach ($DaysArray as $day) {
+            $classday = new ClassDay;
+            $classday->classes_id = $class->id;
+            $classday->DayHeldOn = $day;
+            $classday->save();
+          }
+        }
```

Comment:
I have no idea logically what you're doing here with splitting times and adding things to temporary arrays based on an arbitrary seasontype === 1. It smelly fishy and looks like you overthought and overcomplicated a problem.

Reply:
There are only two season types, 1 and 2. 1 is weekly, 2 is classes held on specific dates. That's all. I'm storing the checkboxes and then looping through those days, creating a ClassDay for each one.

Reply:
A better option would be to have some sort of table called configuration or options. And there you can have a key => value and assign a season_type key to be the strings 'weekly' and 'other'. Then in the code you can call the configuration table key and look it up

No more hardcoding magic values!

Reply:
This was one of those scenarios where I thought there would be more values. Then thinking about there’s really only two options. I suppose you could technically have an every other week or once a month or something dumb like that. But even with those the date-specific option would cover it.